gradoop-data-integration/src/main/java/org/gradoop/dataintegration/transformation/VertexToEdge.java

```diff
@@ -0,0 +1,104 @@
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.operators.neighborhood.keyselector.IdInTuple;
+
+import java.util.List;
+
+
+/**
+ * For a given vertex label this graph transformation takes all neighbours per vertex with this
+ * label and calculates the transitive closure for this subgraph. An edge is created between all
+ * vertice pairs that fulfill the transitive closure requirement.
+ * Furthermore each of those created edges contains the labels of the edges to create the
+ * transitive closure and the former properties of the vertex. <br><br>
+ * <p>
+ * Each edge that has to be created results from a path in the graph of the following form: <br>
+ * v_i --e_i,j--> v_j --e_j,k--> v_k <br>
+ * The newly created edge goes from: v_i --e_i,k--> v_k <br>
+ * The edge {@code e_i,k} has a user-defined label and besides the original vertex properties three
+ * additional properties:
+ * <ul>
+ *   <li>{@code originalVertexLabel}</li>
+ *   <li>{@code firstEdgeLabel = labelOf(e_i,j)}</li>
+ *   <li>{@code secondEdgeLabel = labelOf(e_j,k)}</li>
+ * </ul>
+ */
+public class VertexToEdge implements UnaryGraphToGraphOperator {
+
+  /**
+   * The vertex label of {@code v_j}.
+   */
+  private final String centralVertexLabel;
+
+  /**
+   * The edge label for new edges.
+   */
+  private final String newEdgeLabel;
+
+  /**
+   * The constructor of the operator to transform vertices into edges.
+   *
+   * @param centralVertexLabel The vertex label of {@code v_j}.
+   * @param newEdgeLabel The edge label for new edges.
+   */
+  public VertexToEdge(String centralVertexLabel, String newEdgeLabel) {
+    this.centralVertexLabel = centralVertexLabel;
+    this.newEdgeLabel = newEdgeLabel;
+  }
+
+  @Override
+  public LogicalGraph execute(LogicalGraph graph) {
+    DataSet<Tuple2<Vertex, List<Neighborhood.VertexPojo>>> incomingNeighborhood = Neighborhood
+        .getPerVertex(graph,
+          graph.getVerticesByLabel(centralVertexLabel),
+          Neighborhood.EdgeDirection.INCOMING);
+
+    DataSet<Tuple2<Vertex, List<Neighborhood.VertexPojo>>> outgoingNeighborhood = Neighborhood
+        .getPerVertex(graph,
+          graph.getVerticesByLabel(centralVertexLabel),
+          Neighborhood.EdgeDirection.OUTGOING);
+
+    if (incomingNeighborhood == null || outgoingNeighborhood == null) {
+      throw new IllegalStateException("At least one calculated neighborhood was null.");
+    }
+
+    DataSet<Edge> newEdges = incomingNeighborhood
+        .coGroup(outgoingNeighborhood)
+        .where(new IdInTuple<>(0))
+        .equalTo(new IdInTuple<>(0))
+        .with(new EdgesFromLocalTransitiveClosure(newEdgeLabel,
+          graph.getConfig().getEdgeFactory()));
+
+    return graph.getConfig().getLogicalGraphFactory()
+        .fromDataSets(graph.getVertices(),
+          graph.getEdges().union(newEdges));
+  }
+
+  @Override
+  public String getName() {
+    return this.getClass().getName();
```

Comment:
```suggestion
    return VertexToEdge.class.getName();
```

Reply:
What is the advantage of your suggestion?

Reply:
Unification with existing code. Search the gradoop repo for regex: `return.*\.getName\(\)\;` and you will see that we are using the class itseld everywhere in gradoop. Else, this (and the other two occurrences) will be the first usage of this.getClass() in the whole project. No advantage but uniform.

Reply:
I think we should make
```java
return getClass().getName()
```
the default implementation in the interface. But this should be discussed in another issue.

Reply:
Right! And most important: We have to use this method somewhere XD It's never used atm